packages/agent/test/agent.test.ts

```diff
@@ -31,14 +30,18 @@ describe("@bigtest/agent", function() {
 
   let World: Context;
   function spawn<T>(operation: Operation): Promise<T> {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
     return World["spawn"](operation);
   }
 
   function launch(browserType: BrowserType, options = {}): Promise<Browser> {
     return spawn(({ resume, fail, context: { parent } }) => {
       browserType.launch(options)
         .then(browser => {
```

Comment:
This is easily rewritten as a generator and would improve typing.

Reply:
Can you help me rewrite this?

Reply:
I honestly don't understand why I tried to be clever in the first place
```typescript
  function launch(browserType: BrowserType, options = {}): Promise<Browser> {
    return spawn(function*() {
      let browser: Browser = yield browserType.launch(options);
      try {
        yield;
      } finally {
        browser.close();
      }
    });
  }
```